tensorwaves/physics/helicityformalism/amplitude.py

```diff
@@ -385,22 +385,24 @@ def _create_sequential_amplitude(
     )
 
 
-_AngularProperties = namedtuple(
-    "_AngularProperties", ["j", "m", "mprime", "theta_name", "phi_name"],
-)
+class _AngularProperties(NamedTuple):
+    j: int
+    m: int
+    mprime: int
+    theta_name: str
+    phi_name: str
 
-DynamicsProperties = namedtuple(
-    "DynamicsProperties",
-    [
-        "orbit_angular_momentum",
-        "resonance_mass",
-        "resonance_width",
-        "inv_mass_name",
-        "inv_mass_name_prod1",
-        "inv_mass_name_prod2",
-        "meson_radius",
-    ],
-)
+
+class DynamicsProperties(NamedTuple):
+    """Data structure representing dynamic properties."""
+
+    orbit_angular_momentum: float
+    resonance_mass: float
+    resonance_width: float
+    inv_mass_name: str
+    inv_mass_name_prod1: str
+    inv_mass_name_prod2: str
+    meson_radius: Optional[float]
```

Comment:
@spflueger The key of this PR is whether you agree with the types here (that's one of the advantages of using `typing.NamedTuple` over `collections.namedtuple`). I wasn't 100% here as I didn't write this module.

Reply:
looks good to me